contracts/utils/cryptography/ERC7803Utils.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.24;
+
+import {MessageHashUtils} from "./MessageHashUtils.sol";
+import {Strings} from "../Strings.sol";
+import {Bytes} from "../Bytes.sol";
+import {Math} from "../math/Math.sol";
+
+/**
+ * @dev Utilities to process https://ercs.ethereum.org/ERCS/erc-7803[ERC-7803] signatures with signing domains
+ * for Account Abstraction.
+ *
+ * This library provides methods to encode and decode ERC-7803 signatures that support signing domains
+ * and authentication method coordination. Signing domains prevent replay attacks when private keys
+ * are shared across smart contract accounts, while authentication methods allow dapps and wallets
+ * to coordinate on signature verification approaches.
+ *
+ * The core functionality implements the recursive encoding scheme defined in ERC-7803:
+ *
+ * * If signing domain separators exist: `"\x19\x02" ‖ first ‖ encodeForSigningDomains(others, verifyingDomainSeparator, message)`
+ * * If no signing domain separators: standard EIP-712 encoding
+ */
+library ERC7803Utils {
+    /**
+     * @dev Encodes message for signing domains according to ERC-7803 specification.
+     *
+     * This implements the recursive encoding scheme:
+     *
+     * * If `signingDomainSeparators` is not empty: `"\x19\x02" ‖ first ‖ encodeForSigningDomains(others, verifyingDomainSeparator, message)`
+     * * If `signingDomainSeparators` is empty: standard EIP-712 encoding using {MessageHashUtils-toTypedDataHash}
+     */
+    function encodeForSigningDomains(
+        bytes32[] memory signingDomainSeparators,
+        bytes32 verifyingDomainSeparator,
+        bytes32 structHash
+    ) internal pure returns (bytes32) {
+        return
+            signingDomainSeparators.length == 0
+                ? MessageHashUtils.toTypedDataHash(verifyingDomainSeparator, structHash)
+                : MessageHashUtils.toSigningDomainHash(
+                    signingDomainSeparators[0],
+                    // TODO: Make iterative?
+                    encodeForSigningDomains(
+                        _splice(signingDomainSeparators, 1, signingDomainSeparators.length),
+                        verifyingDomainSeparator,
+                        structHash
+                    )
+                );
+    }
+
+    /// @dev Checks if an authentication method ID corresponds to ECDSA.
+    function isECDSA(string memory methodId) internal pure returns (bool) {
+        return Strings.equal(methodId, "ECDSA");
```

Comment:
When we know the string, and when the length is <=32, it is cheaper (but arguably less readable) to compare things directly

```suggestion
        return bytes(methodId).length == 5 && bytes5(bytes(methodId)) == 0x4543445341;
```